### DISCUSSION TITLE
Add Shape2D as a visual (like Polygon2D) equivalent for existing CollisionShape2D shapes

### DISCUSSION THREAD

**Discussion by omniuni** [godotengine/godot-proposals#10762] [2D]
Based on other areas of Godot and the editor itself, it should be possible to allow Polygon2D to draw the other polygons, such as those available in CollisionShape2D. The engine is fully able to render these shapes in debug mode, so it should be relatively trivial to allow them to be a visual item.

Ideally, Shape2D should include all of the shapes available in CollisionShape2D, allowing it to also potentially replace Polygon2D. This means shapes like Rectangle, Circle, Capsule, and Segment at least.

Minimally, in additional to color and gradient fills, outline color and width should be added as options as well.

Future additions could include toggles for scaling the outline and fill (gradient) when the shape itself is scaled.

It would also be very cool to add a toggle for whether the fill follows rotation or not, allowing for the creation of vector shapes that keep the gradient oriented to create an illusion of shadow, for example.

Polygon2D already allows nesting of other Polygon2D, so this would start to create a foundation of vector graphics within Godot.

Although adding more detailed vector items such as Bezier curves would be a huge lift, these basic shapes are already part of Godot's rendering pipeline, so adding them should be (relatively) trivial.

It would also be nice to allow these shapes to have a physics toggle, OR to enable these same visual options for CollisionShape2D.

Although simple, I believe this could begin to add powerful drawing capabilities that have the potential to grow to a point of enabling some incredible vector-based games in Godot!

**Comment by AThousandShips** [comment]
Thank you for your proposal, consolidating in:
* https://github.com/godotengine/godot-proposals/issues/1126

See there for more context and add any details you feel are missing (and to avoid unnecessary work writing a proposal try searching for existing ones first)

**Reply by omniuni** [reply]
Thank you very much. I had searched, and somehow did not find that proposal. I will copy the remaining parts to that thread.
